database/engine/global_uuid_map/global_uuid_map.c

```diff
@@ -4,17 +4,32 @@
 
 static Pvoid_t JGUID_map = (Pvoid_t) NULL;
 static Pvoid_t JGUID_object_map = (Pvoid_t) NULL;
+static uv_rwlock_t guid_lock;
+static uv_rwlock_t object_lock;
+static uv_rwlock_t global_lock;
 
-int guid_store(uuid_t uuid, char *object)
+
+void guid_bulk_load_lock()
 {
-#ifdef NETDATA_INTERNAL_CHECKS
-    static uint32_t count = 0;
-#endif
-    if (unlikely(!object))
+    uv_rwlock_wrlock(&global_lock);
+
+}
+
+void guid_bulk_load_unlock()
+{
+    uv_rwlock_wrunlock(&global_lock);
+}
+
+
+static inline int guid_store_internal(uuid_t uuid, char *object, int lock)
```

Comment:
It would be good to match the existing style / structure in the dbegine. I think that the convention currently would be to have the internal call called `_nolock` and put the locking in the external wrapper.